### PR TITLE
Bluetooth: Kconfig improvements

### PR DESCRIPTION
**Pull request by KAGA164**
This PR adds a few improvements to Bluetooth subsystem Kconfig:
-  adds name for choices which don't have it
- adds log template for BAS service
- moves Extended Advertising and Isochronous Channel configuration into separated Kconfig files

### CODE REVIEW

File: subsys/bluetooth/Kconfig.iso [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Bluetooth Isochronous Channel configuration options
+
+# Copyright (c) 2016 Intel Corporation
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config BT_ISO
+	bool # hidden
```
Comment (tejlmand) [inline subsys/bluetooth/Kconfig.iso:8]: yes, a promptless symbol is hidden, so no need to state that here.

If you want to make a comment, then make it a proper comment explaining why this symbol is hidden and describe for what it is used.

For example in the help text, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/e1481fee4aa1e61ccef8e64e77aa347b6dd10493/modules/mbedtls/Kconfig#L8-L14

But as minimum remove the comment as it provides no value or info to the reader.
```suggestion
	bool
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
As you are anyway moving Kconfig code around, then it would be nice if you took the time to also align them.

Some parts look like:
```
if BT_EXT_ADV
config BT_EXT_ADV_LEGACY_SUPPORT
```
and
```
endif # BT_ISO_BROADCAST
endif # BT_ISO
```
other parts have an extra empty line
```
if BT_ISO

config BT_ISO_MAX_CHAN
```
and
```
 used for either transmitting or receiving.

endif # BT_ISO_UNICAST
```
and some are mixed:
```
help
	  Maximum number of simultaneous periodic advertising syncs supported.
endif # BT_PER_ADV_SYNC

endif # BT_EXT_ADV
```

I'm only asking to have some consistency in those two files you are creating now.

File: subsys/bluetooth/Kconfig.iso (hunk 1 above)
Comment (KAGA164) [reply subsys/bluetooth/Kconfig.iso]: Done. I just removed comment there. You can easily guess what this option is for, so I didn't add a comment under

File: subsys/bluetooth/Kconfig [hunk 2]
```diff
@@ -117,61 +117,7 @@ config BT_LIM_ADV_TIMEOUT
...
-config BT_PER_ADV_SYNC_MAX
-	int "Maximum number of simultaneous periodic advertising syncs"
-	range 1 64
-	default 1
-	help
-	  Maximum number of simultaneous periodic advertising syncs supported.
-endif # BT_PER_ADV_SYNC
-
-endif # BT_EXT_ADV
+rsource "Kconfig.ext_adv"
```
Comment (joerchan) [inline subsys/bluetooth/Kconfig:120]: There are some `BT_PER_ADV` in what you moved to Kconfig.ext_adv, maybe just move the rest of the adv section there as well (`BT_LIM_ADV_TIMEOUT`) and call it Kconfig.adv?

Comment (KAGA164) [reply subsys/bluetooth/Kconfig:120]: I was thinking about it. However `BT_PER_ADV` depends on `BT_EXT_ADV` but I like an idea to just move all related advertising configs to one file. Will do

Comment (KAGA164) [reply subsys/bluetooth/Kconfig]: Done

**Review by grochu — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

File: subsys/bluetooth/services/Kconfig.bas [hunk 3]
```diff
@@ -7,17 +7,10 @@ menuconfig BT_BAS
...
-	  2 WARNING, write LOG_WRN in addition to previous level
-	  3 INFO, write LOG_INF in addition to previous levels
-	  4 DEBUG, write LOG_DBG in addition to previous levels
+if BT_BAS
+
+module = BT_BAS
+module-str = BAS
+source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
+
+endif # BT_BAS
```
Comment (Thalley) [inline subsys/bluetooth/services/Kconfig.bas:16]: I'm not too familiar with this: Is the `CONFIG_BT_BAS_LOG_LEVEL` still defined after this change?

Comment (KAGA164) [reply subsys/bluetooth/services/Kconfig.bas:16]: Yes it is defined by this template. There isn't need to change a `bas` service

**Review by Thalley — APPROVED** [review]

**Comment by KAGA164** [comment]
@carlescufi @jhedberg @Vudentz Can we merge this ?

**Review by doki-nordic — APPROVED** [review]

**Pull request merged**